pkg/pool-utils/test/AumProtocolFeeCache.test.ts

```diff
@@ -0,0 +1,155 @@
+    await aumProtocolFeesCollector.connect(admin).setAumFeePercentage(AUM_PROTOCOL_FEE);
+  });
+
+  context('with delegated swap fee', () => {
+    sharedBeforeEach('deploy delegated fee cache', async () => {
+      await vaultObj.setSwapFeePercentage(VAULT_PROTOCOL_FEE, { from: admin });
+
+      // The sentinel value used to designate delegated fees is MAX_UINT256
+      protocolFeeCache = await deploy('MockAumProtocolFeeCache', {
+        args: [vault.address, MAX_UINT256, aumProtocolFeesCollector.address],
+      });
+    });
+
+    it('indicates delegated fees', async () => {
+      expect(await protocolFeeCache.getProtocolFeeDelegation()).to.be.true;
+    });
+
+    it('gets the protocol fee from the vault', async () => {
+      expect(await protocolFeeCache.getProtocolSwapFeePercentageCache()).to.equal(VAULT_PROTOCOL_FEE);
+    });
+
+    context('when the vault fee is updated', () => {
+      sharedBeforeEach('update the main protocol fee', async () => {
+        await vaultObj.setSwapFeePercentage(NEW_VAULT_PROTOCOL_FEE, { from: admin });
+      });
+
+      it('retrieves the old value when not updated', async () => {
+        expect(await protocolFeeCache.getProtocolSwapFeePercentageCache()).to.equal(VAULT_PROTOCOL_FEE);
+      });
+
+      it('updates the cached value', async () => {
+        await protocolFeeCache.updateProtocolSwapFeePercentageCache();
+
+        expect(await protocolFeeCache.getProtocolSwapFeePercentageCache()).to.equal(NEW_VAULT_PROTOCOL_FEE);
+      });
+
+      it('emits an event when updating', async () => {
+        const receipt = await protocolFeeCache.updateProtocolSwapFeePercentageCache();
+
+        expectEvent.inReceipt(await receipt.wait(), 'ProtocolSwapFeePercentageCacheUpdated', {
+          protocolSwapFeePercentage: NEW_VAULT_PROTOCOL_FEE,
+        });
+      });
+    });
+  });
+
+  context('with fixed swap fee', () => {
+    sharedBeforeEach('deploy fixed fee cache', async () => {
+      protocolFeeCache = await deploy('MockAumProtocolFeeCache', {
+        args: [vault.address, FIXED_PROTOCOL_FEE, aumProtocolFeesCollector.address],
+      });
+    });
+
+    it('indicates fixed fees', async () => {
+      expect(await protocolFeeCache.getProtocolFeeDelegation()).to.be.false;
+    });
+
+    it('sets the protocol fee', async () => {
+      expect(await protocolFeeCache.getProtocolSwapFeePercentageCache()).to.equal(FIXED_PROTOCOL_FEE);
+    });
+
+    it('reverts if fee is too high', async () => {
+      await expect(
+        deploy('MockAumProtocolFeeCache', {
+          args: [vault.address, MAX_PROTOCOL_FEE.add(1), aumProtocolFeesCollector.address],
+        })
+      ).to.be.revertedWith('SWAP_FEE_PERCENTAGE_TOO_HIGH');
+    });
+
+    it('reverts when trying to update fixed fee', async () => {
+      await expect(protocolFeeCache.updateProtocolSwapFeePercentageCache()).to.be.revertedWith('INVALID_OPERATION');
+    });
+  });
```

Comment:
I don't think we need to repeat all of the protocol swap fee cache tests here. If you really want to do it, they should be abstracted away into a behavior and reused that way.

Reply:
Just removing for now, then. There are no other tests in the base class, so the base class test would just be one line calling the behavior.